### PR TITLE
✨ Add topology field for NodeDeletionTimeout

### LINKED ISSUE #6450: ClusterClass make MachineDeployment NodeDeletionTimeout configurable via Cluster topology
**User Story**

As a user I would like to configure the NodeDeletionTimeout of a MachineDeployment.

**Details**

Prior art for this is the addition of nodeDrainTimeout in #6278.

PR: https://github.com/kubernetes-sigs/cluster-api/pull/6379
Detailed description on what to do: https://github.com/kubernetes-sigs/cluster-api/issues/6278#issuecomment-1070914169 (of course only the NodeDeletionTimeout parts are relevant)

/kind feature
/area topology

### PR DESCRIPTION
**Pull request by abhijit-dev82**
**What this PR does / why we need it**:
Make NodeDeletionTimeOut field in MachineDeployment configurable.

**Which issue(s) this PR fixes** 
Fixes #6450

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
/ok-to-test

**Comment by killianmuldoon** [comment]
@abhijit-dev82 could you run `make generate` on this to pass the verify job?

**Comment by abhijit-dev82** [comment]
@killianmuldoon  had missed the open-api generated file. Have added it now, the checks should go through.

File: internal/contract/controlplane_test.go [hunk 1]
```diff
@@ -170,6 +170,28 @@ func TestControlPlane(t *testing.T) {
...
+		got, err := ControlPlane().MachineTemplate().NodeDeletionTimeout().Get(obj)
+		g.Expect(err).ToNot(HaveOccurred())
+		g.Expect(got).ToNot(BeNil())
+		g.Expect(*got).To(Equal(duration))
+
+		// Check that the literal string value of the duration is correctly formatted.
+		durationString, found, err := unstructured.NestedString(obj.UnstructuredContent(), "spec", "machineTemplate", "nodeDeletionTimeout")
+		g.Expect(err).ToNot(HaveOccurred())
+		g.Expect(found).To(BeTrue())
+		g.Expect(durationString).To(Equal(expectedDurationString))
```
Comment (killianmuldoon) [inline internal/contract/controlplane_test.go:193]: 👍

**Review by killianmuldoon — APPROVED** [review]
/lgtm

Thanks for this @abhijit-dev82!

**Comment by sbueringer** [comment]
Thank you very much!

/approve

**Pull request merged**